versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java

```diff
@@ -391,19 +394,16 @@ public boolean storeObj(
       ObjTypes.allObjTypes().stream()
           .collect(
               ImmutableMap.toImmutableMap(
-                  Function.identity(), (ObjType type) -> ByteString.copyFromUtf8(type.name())));
+                  Function.identity(), (ObjType type) -> copyFromUtf8(type.name())));
 
-  @NotNull
+  @Nonnull
   private ConditionalRowMutation mutationForStoreObj(
-      @NotNull Obj obj, boolean ignoreSoftSizeRestrictions) throws ObjTooLargeException {
+      @Nonnull Obj obj, @Nonnull PersistOptions options) throws ObjTooLargeException {
     checkArgument(obj.id() != null, "Obj to store must have a non-null ID");
     ByteString key = dbKey(obj.id());
 
-    int incrementalIndexSizeLimit =
-        ignoreSoftSizeRestrictions ? Integer.MAX_VALUE : effectiveIncrementalIndexSizeLimit();
-    int indexSizeLimit =
-        ignoreSoftSizeRestrictions ? Integer.MAX_VALUE : effectiveIndexSegmentSizeLimit();
-    byte[] serialized = serializeObj(obj, incrementalIndexSizeLimit, indexSizeLimit);
+    SizeLimits limits = options.ignoreSoftSizeRestrictions() ? SizeLimits.NO_LIMITS : defaultLimits;
```

Comment:
This is the same computation for all calls to `serializeObj`. Better move into `serializeObj` and make it non-static